### DISCUSSION TITLE
Help with fixing malformed 'albumtype' and 'albumtypes' fields

### DISCUSSION THREAD

**Discussion by chrisbenincasa** [beetbox/beets#4928] [Q&A] [answered]
Not sure what happened here, but every one of my `albumtypes` and `albumtype` fields got malformed. Example:

```
beet info -i albumtypes,albumtype /data/audio/ABBA/The Visitors (1981)/ABBA - The Visitors - 03 - When All Is Said and Done.flac

/data/audio/ABBA/The Visitors (1981)/ABBA - The Visitors - 03 - When All Is Said and Done.flac
 albumtype: [
albumtypes: [; '; a; '; ,;  ; '; l; '; ,;  ; '; b; '; ,;  ; '; u; '; ,;  ; '; m; '; ]
```

When viewing the raw tags using ffprobe:

```
Input #0, flac, from '/data/audio/ABBA/The Visitors (1981)/ABBA - The Visitors - 03 - When All Is Said and Done.flac':
  Metadata:
    ALBUM           : The Visitors
    ALBUM ARTIST    : ABBA
    album_artist    : ABBA
    ALBUMARTIST_CREDIT: ABBA
    ALBUMARTISTSORT : ABBA
    RELEASESTATUS   : Official
    MUSICBRAINZ_ALBUMSTATUS: Official
    RELEASETYPE     : [;';a;';,; ;';l;';,; ;';b;';,; ;';u;';,; ;';m;';]
    MUSICBRAINZ_ALBUMTYPE: [;';a;';,; ;';l;';,; ;';b;';,; ;';u;';,; ;';m;';]
```

Clearly there was some weird string <-> list splitting/joining issue...somewhere. I'm wondering if there is a reasonable way to correct the tags for all of the files, without going through doing laboriously retagging everything. The "right" info seems to be present...just incorrectly encoded. Any ideas??

**Comment by sampsyo** [comment]
This looks reminiscent of #4528… is there any chance that this goes away if you switch to the current git version of beets?

**Reply by chrisbenincasa** [reply]
Something later than 1.6.0? Like HEAD?

FWIW I am running 1.6.0 now
```
media@beets:~$ beet version
beets version 1.6.0
Python version 3.9.2
```

**Reply by sampsyo** [reply]
Yes, exactly—the current `master` branch.

**Reply by chrisbenincasa** [reply]
I have beets install via pip. Is there a non-destructive way to install the newer version side-by-side (if i want to keep the stable release but test some things on the edge?)

**Reply by chrisbenincasa** [reply]
I did just try cloning the repo and running directly...but some modules are missing. Let me try something else, actually...

**Reply by sampsyo** [reply]
There is an FAQ about this:
https://beets.readthedocs.io/en/stable/faq.html#run-the-latest-source-version-of-beets

**Reply by chrisbenincasa** [reply]
Ah ok, thanks. I'll do this. I was hoping to be able to have 2 different versions live side-by-side with the known good being the "main" one.

**Reply by chrisbenincasa** [reply]
Ok, so downloading the latest and retagging an album fixes the issue.

Though I'm now trying to discern if there is a way to fix this without reimporting everything, since everything in the library is corrupted at this point.

**Reply by chrisbenincasa** [reply]
Hmm... OK so I think the problem still persists on HEAD. I edited the DB directly to fix any albumtype/albumtypes fields that had been saved to the db in the "array" format.

So now when querying info, these are being printed correctly:
```
media@beets:~$ beet info -i albumtype,albumtypes -a /data/audio/ABBA/Waterloo\ \(1974\)/
/data/audio/ABBA/Waterloo (1974)
 albumtype: album
albumtypes: album

```

However, when I run either `beet update` or `beet write`, I get erroneous changes. Almost as if the "split" occurring on the DB field is incorrect. Examples:

```
media@beets:~$ beet update -p -F albumtype -F albumtypes artist:ABBA album:Waterloo
ABBA - Waterloo - Waterloo
  albumtype: a -> [
  albumtypes: ['a', 'l', 'b', 'u', 'm'] -> [; '; a; '; ,;  ; '; l; '; ,;  ; '; b; '; ,;  ; '; u; '; ,;  ; '; m; '; ]
ABBA - Waterloo - Sitting in the Palmtree
  albumtype: a -> [
  albumtypes: ['a', 'l', 'b', 'u', 'm'] -> [; '; a; '; ,;  ; '; l; '; ,;  ; '; b; '; ,;  ; '; u; '; ,;  ; '; m; '; ]
// more below
```

and write:
```
media@beets:~$ beet write -p  artist:ABBA album:Waterloo                  
ABBA - Waterloo - Waterloo                                                                                                                         
  mb_artistids: d87e52c5-bb8d-4da8-b941-9f4928627dc8 -> 
  mb_albumartistids: d87e52c5-bb8d-4da8-b941-9f4928627dc8 -> 
  albumtype: [ -> a                                                      
  albumtypes: [; '; a; '; ,;  ; '; l; '; ,;  ; '; b; '; ,;  ; '; u; '; ,;  ; '; m; '; ] -> ['a', 'l', 'b', 'u', 'm']
ABBA - Waterloo - Sitting in the Palmtree                                                                                                          
  mb_artistids: d87e52c5-bb8d-4da8-b941-9f4928627dc8 -> 
  mb_albumartistids: d87e52c5-bb8d-4da8-b941-9f4928627dc8 -> 
  albumtype: [ -> a                                                      
  albumtypes: [; '; a; '; ,;  ; '; l; '; ,;  ; '; b; '; ,;  ; '; u; '; ,;  ; '; m; '; ] -> ['a', 'l', 'b', 'u', 'm']
// more below
```

**Reply by JOJ0** [reply, marked as answer]
This might help @chrisbenincasa 

https://github.com/beetbox/beets/pull/4582#issuecomment-1445023493
